monitor-manager/monitor-manager.py

```diff
@@ -0,0 +1,76 @@
+from csv import reader
+from flask import Flask
+import socket
+import json
+import requests
+import time
+
+app = Flask(__name__)
+
+
+@app.route('/', methods=['get'])
+def health_check():
+    return "Monitor Manager working"
+
+
+@app.route('/experiments', methods=['post'])
+def start_experiments():
+    experiments_plan = read_csv("experiments-plan.csv")
+    hosts_info = read_csv("hosts-info.csv")
+    is_spark_node, host_ip = check_if_spark_node(hosts_info)
+
+    for function_name, table in experiments_plan:
+        print("Task submitting")
+        submit_response = requests.post("http://192.168.55.20:5000/submit", json={"function_name": function_name})
+        submit_response = json.loads(submit_response.text)
+        driver_id = submit_response["submissionId"]
+        print("Submit driver ID: ", driver_id)
+
+        print("Start monitoring")
+        system_monitor(host_ip, hosts_info, function_name, True)
+        #
+        status = "RUNNING"
+        while status == "RUNNING":
+            status_response = requests.get("http://192.168.55.20:5000/status", json={"driver_id": driver_id})
+            status_response = json.loads(status_response.text)
+            status = status_response["driverState"]
+            time.sleep(1)
+
+        system_monitor(host_ip, hosts_info, function_name, False)
+
+    return "Monitoring finished", 200
+
+
+def system_monitor(host_ip, hosts_info, function_name, start):
+    for ip, container_name in hosts_info:
+        url = "http://" + str(ip) + ":8063/monitor"
+        request_data = {
+            "container_name": container_name,
+            "function_name": function_name
+        }
+        if start:
+            response = requests.post(url, json=request_data)
+            print(response.text)
+        else:
+            response = requests.delete(url)
+        print("IP: ", ip, " ", response.text)
+
+
+def read_csv(filename):
+    with open(filename, 'r') as file:
+        file_reader = reader(file, delimiter=',')
+        header = next(file_reader)
+        return list(file_reader)
+
+
+def check_if_spark_node(hosts_info):

```

Comment:
Is really necessary? What does it change on which node we are?

Reply:
Now you run this python script on local machine. This function is added in case we want to run on one of our nodes on cluster.
If the script is ran on host outside the cluster, then post requests to start python monitor are send to the url with host ip address in it, like _http://192.168.55.12:8063_.
In case this script is ran on master node then this url should be changed to _http://localhost:8063_
This function is now not used, because we decided that this script will be ran on local machine, but at first i implemented mechanism which used it.

Reply:
I'm not sure, but is there any difference between calling your local machine with its own IP and calling localhost?